I tried this but it's hitting static assertions...

```

void Cursor::updateBuffer(const uint8_t* pixelsData, uint32_t width, uint32_t height, uint32_t stride)
{
    RELEASE_ASSERT(width <= m_deviceWidth);
    RELEASE_ASSERT(height <= m_deviceHeight);

    std::span<const uint8_t> pixels { pixelsData, width * height * stride };
    Vector<uint32_t> deviceBuffer(m_deviceWidth * m_deviceHeight);
    size_t offset = 0;
    size_t pixelsOffset = 0;
    for (uint32_t i = 0; i < height; ++i) {
        memcpySpan(deviceBuffer.subspan(offset, stride), pixels.subspan(pixelsOffset, stride));
        offset += i * m_deviceWidth;
        pixelsOffset += i * stride;
    }
    gbm_bo_write(m_buffer->bufferObject(), deviceBuffer.data(), deviceBuffer.sizeInBytes());
}
```
/var/home/phil/WebKit/WebKitBuild/WPE/Release/WTF/Headers/wtf/StdLibExtras.h:706:19: error: static assertion failed due to requirement 'sizeof(const unsigned int) == sizeof(const unsigned char)'                 
  706 |     static_assert(sizeof(T) == sizeof(U));                                                                                                                                                                 
      |                   ^~~~~~~~~~~~~~~~~~~~~~

--- a/Source/WebCore/platform/xr/openxr/PlatformXROpenXR.cpp
+++ b/Source/WebCore/platform/xr/openxr/PlatformXROpenXR.cpp
@@ -366,14 +366,14 @@ void OpenXRDevice::collectSupportedSessionModes()
     auto result = xrEnumerateViewConfigurations(m_instance, m_systemId, 0, &viewConfigurationCount, nullptr);
     RETURN_IF_FAILED(result, "xrEnumerateViewConfigurations", m_instance);
 
-    XrViewConfigurationType viewConfigurations[viewConfigurationCount];
-    result = xrEnumerateViewConfigurations(m_instance, m_systemId, viewConfigurationCount, &viewConfigurationCount, viewConfigurations);
+    Vector<XrViewConfigurationType> viewConfigurations(viewConfigurationCount);
+    result = xrEnumerateViewConfigurations(m_instance, m_systemId, viewConfigurationCount, &viewConfigurationCount, viewConfigurations.data());
     RETURN_IF_FAILED(result, "xrEnumerateViewConfigurations", m_instance);
 
     FeatureList features = collectSupportedFeatures();
-    for (uint32_t i = 0; i < viewConfigurationCount; ++i) {
+    for (auto& viewConfiguration : viewConfigurations) {
         auto viewConfigurationProperties = createStructure<XrViewConfigurationProperties, XR_TYPE_VIEW_CONFIGURATION_PROPERTIES>();
-        result = xrGetViewConfigurationProperties(m_instance, m_systemId, viewConfigurations[i], &viewConfigurationProperties);
+        result = xrGetViewConfigurationProperties(m_instance, m_systemId, viewConfiguration, &viewConfigurationProperties);
         if (result != XR_SUCCESS) {
             LOG(XR, "xrGetViewConfigurationProperties(): error %s\n", resultToString(result, m_instance).utf8().data());
             continue;

--- a/Source/WebKit/WPEPlatform/wpe/drm/WPEDRMCursor.cpp
+++ b/Source/WebKit/WPEPlatform/wpe/drm/WPEDRMCursor.cpp
@@ -69,11 +69,10 @@ void Cursor::updateBuffer(const uint8_t* pixels, uint32_t width, uint32_t height
     RELEASE_ASSERT(width <= m_deviceWidth);
     RELEASE_ASSERT(height <= m_deviceHeight);
 
-    uint32_t deviceBuffer[m_deviceWidth * m_deviceHeight];
-    memset(deviceBuffer, 0, sizeof(deviceBuffer));
+    Vector<uint32_t> deviceBuffer(m_deviceWidth * m_deviceHeight);
     for (uint32_t i = 0; i < height; ++i)
-        memcpy(deviceBuffer + i * m_deviceWidth, pixels + i * stride, stride);
-    gbm_bo_write(m_buffer->bufferObject(), deviceBuffer, sizeof(deviceBuffer));
+        memcpy(&deviceBuffer[i * m_deviceWidth], pixels + i * stride, stride);
+    gbm_bo_write(m_buffer->bufferObject(), deviceBuffer.data(), deviceBuffer.sizeInBytes());
 }
 
 void Cursor::setFromName(const char* name, double scale)

